jbmc/src/java_bytecode/select_pointer_type.cpp

```diff
@@ -109,3 +135,95 @@ pointer_typet select_pointer_typet::specialize_generics(
   }
   return pointer_type;
 }
+
+/// Return the first concrete type instantiation if any such exists.
```

Comment:
If this function is meant to only be used to break cycles in specialization, please make it clear in documentation and an example would be very helpful here too.

Reply:
I added
```
/// Return the first concrete type instantiation if any such exists. This method
/// is only to be called when the `specialize_generics` cannot find an
/// instantiation due to a loop in its recursion.
```